cognite/neat/rules/models/_base_rules.py

```diff
@@ -348,3 +350,13 @@ def __getitem__(self, index: SupportsIndex | slice, /) -> T_SheetRow | SheetList
     ),
     BeforeValidator(lambda v: ExtensionCategory(v) if isinstance(v, str) else v),
 ]
+
+
+# Immutable such that this can be used as a key in a dictionary
+class PropertyRef(BaseModel, frozen=True):
+    class_: ClassEntityType = Field(alias="Class")
+    property_: InformationPropertyType = Field(alias="Property")
+
+
+class ClassRef(BaseModel, frozen=True):
+    class_: ClassEntityType = Field(alias="Class")
```

Comment:
I think these will be useful in more context. Main motivation for adding them to the `_base_rules` is to have the `as_reference()` methods in `InformationProperty` and `InformationClass`.